akka-http-core/src/main/scala/akka/http/impl/engine/rendering/RenderSupport.scala

```diff
@@ -79,13 +79,13 @@ private object RenderSupport {
     override def onPush(elem: ByteString, ctx: Context[ByteString]): SyncDirective = {
       sent += elem.length
       if (sent > length)
-        throw InvalidContentLengthException(s"HTTP message had declared Content-Length $length but entity data stream amounts to more bytes")
+        ctx fail InvalidContentLengthException(s"HTTP message had declared Content-Length $length but entity data stream amounts to more bytes")
```

Comment:
The question is whether these should be raised as an exceptional server condition (and therefore logged) or considered validation errors for client input (and not logged). If the server does everything right then in my opinion there shouldn’t be any log output; in particular (and as an idealized goal) it should not be possible for a remote attacker to control the volume of log messages that are generated. My impression was that while not perfect we are actually quite close to this ideal.